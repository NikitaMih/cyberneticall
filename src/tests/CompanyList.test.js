import { render } from '@testing-library/react';
import CompanyList from '../components/companyList';
import * as reduxHooks from 'react-redux';

jest.mock('react-redux');

const mockedDispatch = jest.spyOn(reduxHooks, 'useDispatch');

describe('CompanyList', () => {
  it('Should create empty CompanyList', () => {
    mockedDispatch.mockReturnValue();
    const view = render(<CompanyList />);

    expect(view).toMatchSnapshot();
  })
});
