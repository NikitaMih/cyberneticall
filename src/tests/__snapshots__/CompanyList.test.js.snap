// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompanyList Should create empty CompanyList 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <table
        class="company-list"
      >
        <thead>
          <tr
            class="company-list__head"
          >
            <th
              class="company-list__name"
            >
              Company name
              <button
                class="company-list__btn"
              >
                <img
                  alt="find"
                  src="find.svg"
                />
              </button>
            </th>
            <th>
              Region
            </th>
            <th>
              Currency
            </th>
            <th />
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </body>,
  "container": <div>
    <table
      class="company-list"
    >
      <thead>
        <tr
          class="company-list__head"
        >
          <th
            class="company-list__name"
          >
            Company name
            <button
              class="company-list__btn"
            >
              <img
                alt="find"
                src="find.svg"
              />
            </button>
          </th>
          <th>
            Region
          </th>
          <th>
            Currency
          </th>
          <th />
        </tr>
      </thead>
      <tbody />
    </table>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
